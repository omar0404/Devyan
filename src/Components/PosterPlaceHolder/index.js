import PosterPlaceHodler from './PosterPlaceHolder';
export default PosterPlaceHodler;
