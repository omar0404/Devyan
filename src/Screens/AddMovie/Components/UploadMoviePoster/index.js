import UploadMoviePoster from './UploadMoviePoster';
export default UploadMoviePoster;
