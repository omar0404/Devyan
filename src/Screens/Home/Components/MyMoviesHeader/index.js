import MyMoviesHeader from './MyMoviesHeader';
export default MyMoviesHeader;
