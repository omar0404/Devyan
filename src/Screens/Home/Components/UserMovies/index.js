import UserMovies from './UserMovies';
export default UserMovies;
