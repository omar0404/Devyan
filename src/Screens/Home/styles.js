import {StyleSheet} from 'react-native';
import theme from 'Devyan/src/Theme';
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 15,
    backgroundColor: theme.backgroundColor,
  },
});
export default styles;
