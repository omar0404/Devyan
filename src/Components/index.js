import Section from './Section';
import PosterPlaceHolder from './PosterPlaceHolder';
import MovieCard from './MovieCard';
export {MovieCard, Section, PosterPlaceHolder};
