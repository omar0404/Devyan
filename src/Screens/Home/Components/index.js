import MyMovies from './MyMovies';
import AllMovies from './AllMovies';
import UserMovies from './UserMovies';
import MyMoviesHeader from './MyMoviesHeader';
export {MyMovies, AllMovies, UserMovies, MyMoviesHeader};
