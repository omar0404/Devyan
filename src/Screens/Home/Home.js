import React, {useState} from 'react';
import {FlatList} from 'react-native';
import {MyMovies, AllMovies} from './Components';
import styles from './styles';
import {useNavigation} from '@react-navigation/native';
const Home = () => {
  const navigation = useNavigation();
  const [likedMovies, setLikedMovies] = useState({});

  const handleUpdateLikedMovies = (movie) => {
    const movieId = movie.id;
    if (likedMovies[movieId]) {
      delete likedMovies[movieId];
      setLikedMovies((prevLikedMovies) => ({...prevLikedMovies}));
      return;
    }
    setLikedMovies((prevLikedMovies) => ({
      ...prevLikedMovies,
      [movieId]: movie,
    }));
  };
  const handleMoviePress = (movie) => {
    navigation.navigate('Movie', {movie});
  };
  return (
    <>
      <FlatList
        style={styles.container}
        ListHeaderComponent={
          <MyMovies
            onMoviePress={handleMoviePress}
            movies={likedMovies}
            onUnLikePress={handleUpdateLikedMovies}
          />
        }
        ListFooterComponent={
          <AllMovies
            onMoviePress={handleMoviePress}
            likedMovies={likedMovies}
            onLikePress={handleUpdateLikedMovies}
          />
        }
      />
    </>
  );
};
export default Home;
