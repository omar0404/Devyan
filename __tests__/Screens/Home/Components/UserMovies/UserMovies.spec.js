import React from 'react';
import {create} from 'react-test-renderer';
import {UserMovies} from 'MoviesApp/src/Screens/Home/Components';
import {shallow} from 'enzyme';

describe('UserMovies Component with no movies', () => {
  it('should render no movies added text ', () => {
    const wrapper = shallow(<UserMovies />);
    expect(wrapper.find({testID: 'noMoviesAdded'}).props().children).toBe(
      'No Movies Added',
    );
  });
});

describe('UserMovies Component with movies ', () => {
  it('should render user movies  ', () => {
    const movies = [{name: 'Batman'}, {name: 'SuperMan'}];
    const wrapper = create(<UserMovies movies={movies} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
