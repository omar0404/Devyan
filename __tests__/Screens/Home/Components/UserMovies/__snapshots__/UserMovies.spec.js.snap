// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMovies Component with movies  should render user movies   1`] = `
<RCTScrollView
  horizontal={true}
  style={
    Object {
      "marginTop": 10,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 200,
          "marginRight": 20,
          "opacity": 1,
          "width": 200,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "gray",
              "borderWidth": 1,
              "flex": 2,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "gray",
                "fontSize": 50,
              },
              undefined,
              Object {
                "fontFamily": "Entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginTop": 12,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "white",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgb(255, 255, 255)",
                "flex": 1,
              }
            }
          >
            <Text>
               
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#991F25",
                  "flex": 1,
                  "fontWeight": "bold",
                  "textAlign": "right",
                }
              }
            />
          </Text>
          <Text
            allowFontScaling={false}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "red",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          numberOfLines={3}
          style={
            Object {
              "color": "gray",
              "marginTop": 12,
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 200,
          "marginRight": 20,
          "opacity": 1,
          "width": 200,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "gray",
              "borderWidth": 1,
              "flex": 2,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "gray",
                "fontSize": 50,
              },
              undefined,
              Object {
                "fontFamily": "Entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginTop": 12,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "white",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgb(255, 255, 255)",
                "flex": 1,
              }
            }
          >
            <Text>
               
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#991F25",
                  "flex": 1,
                  "fontWeight": "bold",
                  "textAlign": "right",
                }
              }
            />
          </Text>
          <Text
            allowFontScaling={false}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "red",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          numberOfLines={3}
          style={
            Object {
              "color": "gray",
              "marginTop": 12,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
